### ISSUE TITLE
'Stretching' sprites on WebGL Renderer

### ISSUE THREAD

**Issue by themoonrat** [pixijs/pixi.js#5954]
### Expected Behavior
In v4, they look like: 
![image](https://user-images.githubusercontent.com/1528426/61963020-ba8f9900-afc2-11e9-8efc-25a8a95390bb.png)

### Current Behavior
In the latest v5, they look like: 
![image](https://user-images.githubusercontent.com/1528426/61963058-c67b5b00-afc2-11e9-8c21-a60fe073c392.png)
Notice the stretching to the right and the bottom.

texture and render resolutions are below 1.

Wierdly, if I move the right hand sprites to the left, or move the bottom sprites upwards, they no longer show the 'stretched' aspect.

### Environment

- **`pixi.js` version**: 5.1.0
- **Browser & Version**: Chrome 77 & Firefox 69
- **OS & Version**: Windows 10

Seeing if I can break this down into an example now

**Comment by themoonrat** [comment]
Ok, so I can't replicate on a playground, but I can find where things go wrong.

So the resolution of the game is 0.5, and assets loaded in are 0.5. I tracked down what was going wrong at `this.resolution = resolution || settings.RESOLUTION` in BaseTexture.js. Basically, if I hard coded the `|| settings.RESOLUTION` to 1, everything worked fine. But if I didn't, I'd get the image stretching as above. 

So I put a breakpoint for every time resolution was undefined coming into this; and one by one found out where the visuals were broken if the resolution for a certain base texture was not 1. And here is the call stack with where I tracked the issue down to
![image](https://user-images.githubusercontent.com/1528426/61969071-896a9500-afd1-11e9-86e7-ca731663c73a.png)

Sprite Masks. `getOptimalTexture` is passed a resolution, but it creates the renderTexture with just a minWidth and minHeight, and THEN tries to set a resolution on it. This doesn't seem to work :/ If I hack it to create the render texture with the correct resolution straight away, everything works correctly

**Comment by themoonrat** [comment]
And so the lazy setting of the resolution is because
`baseTexture.resolution === resolution` in setResolution for RenderTexture. Therefore it returns and nothing happens. I'm not sure what it is in there that needs updating that's not occurring otherwise... but this is the end of my 3 hour journey. Hope someone can help from here!

**Comment by themoonrat** [comment]
If I do something hacky like

```
if (!renderTexture)
{
	renderTexture = this.createTexture(minWidth, minHeight, resolution);
}

 renderTexture.filterPoolKey = key;
renderTexture.setResolution(1);
renderTexture.setResolution(resolution);
```
Then it works. I _must_ create the renderTexture with the correct target resolution to start with (otherwise it gets stretchy stretchy) and I _must_ force the resolution to be something wrong, then back to the right one again (otherwise it gets stretchy stretchy)

**Comment by themoonrat** [comment]
More info. Took a screenshot of the 'broken' version renderTexture used, compared to the hacky fix I created.

Broken
![broken](https://user-images.githubusercontent.com/1528426/61993878-63450380-b06a-11e9-9fc0-03c56ae287e9.png)

Fixed
![fixed](https://user-images.githubusercontent.com/1528426/61993879-663ff400-b06a-11e9-8036-5497083f1ab7.png)

Looks like my fix could be a valid route. Because in getOptimalTexture, minWidth and minHeight have already been multiplied by the resolution of 0.5 to get the correct required width. But then to have that renderTexture be set at 0.5 resolution again is where things are thrown off. So always create at * 1 (tbf, createTexture  does say *real*Width and *real*Height), and then set the resolution after to match
